Source/CarthageKit/Xcode.swift

```diff
@@ -971,31 +975,52 @@ public func buildInDirectory(directoryURL: NSURL, withConfiguration configuratio
 				return false
 			}
 		}
-		|> take(1)
-		|> flatMap(.Merge) { (project: ProjectLocator) -> SignalProducer<String, CarthageError> in
+		|> flatMap(.Concat) { (project: ProjectLocator) -> SignalProducer<[SchemeInProject], CarthageError> in
 			return schemesInProject(project)
+				|> flatMap(.Merge) { scheme -> SignalProducer<SchemeInProject, CarthageError> in
+					let buildArguments = BuildArguments(project: project, scheme: scheme, configuration: configuration)
+
+					return shouldBuildScheme(buildArguments, platform)
+						|> filter { $0 }
+						|> map { _ in SchemeInProject(scheme: scheme, project: project) }
+				}
+				|> collect
+				|> filter { !$0.isEmpty }
 		}
-		|> combineLatestWith(locatorProducer |> take(1))
-		|> map { (scheme: String, project: ProjectLocator) -> BuildSchemeProducer in
-			let buildArguments = BuildArguments(project: project, scheme: scheme, configuration: configuration)
-
-			return shouldBuildScheme(buildArguments, platform)
-				|> filter { $0 }
-				|> flatMap(.Merge) { _ in
-					let initialValue = (project, scheme)
-
-					let buildProgress = buildScheme(scheme, withConfiguration: configuration, inProject: project, workingDirectoryURL: directoryURL)
-						// Discard any existing Success values, since we want to
-						// use our initial value instead of waiting for
-						// completion.
-						|> map { taskEvent in
-							return taskEvent.map { _ in initialValue }
-						}
-						|> filter { taskEvent in taskEvent.value == nil }
+		|> take(1)
+		|> flatMap(.Merge) { schemes in SignalProducer(values: schemes) }
+		|> flatMap(.Merge) { scheme, project -> SignalProducer<(String, ProjectLocator), CarthageError> in
+			return locatorProducer
```

Comment:
Should we avoid enumerating the given directory per scheme? That would be inefficient.